cpp/src/arrow/util/ree_util.h

```diff
@@ -138,6 +152,13 @@ int64_t FindPhysicalIndex(const ArraySpan& span, int64_t i, int64_t absolute_off
 /// end). This function uses binary-search, so it has a O(log N) cost.
 int64_t FindPhysicalLength(const ArraySpan& span);
 
+/// \brief Find the physical range of physical values referenced by the REE in
+/// the logical range from offset to offset + length
+///
+/// \return a pair of physical offset and physical length
+std::pair<int64_t, int64_t> FindPhysicalRange(const ArraySpan& span, int64_t offset,
```

Comment:
Can you add unit tests for this new function?

Reply:
As length calculation is defined in terms of it, I thought tests would be redundant. But I will add them now close to the length tests.